javaslang/src/main/java/javaslang/Value.java

```diff
@@ -1021,6 +1013,26 @@ default CharSeq toCharSeq() {
     @Override
     String toString();
 
+    /**
+     * Tests equality of two objects assuming one, or both might be instances of Value.
+     *
+     * If one of a or b is Value instance congruence is tested.
+     * If neither is Objects{@link Objects#equals} is being invoked for two.
+     *
+     * @param a first object to test
+     * @param b fecond object to test
+     * @return true, if this equals o according to the rules defined above, otherwise false.
+     */
+    static boolean eq(Object a, Object b) {
```

Comment:
Extracting this method is possible but we do not need to introduce new public API.

On the 'level' of Value it is not intended to fall back to `equals`. The `eq(Object)` does only fall back for contained values.